### ISSUE TITLE
[explicit-module-boundary-types] Warning for Return Types and Argument Types on Nested Functions

### ISSUE THREAD

**Issue by ccaspers** [typescript-eslint/typescript-eslint#2925]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

Fully working repro can be found here: https://github.com/ccaspers/explicit-module-boundary-types

```js
module.exports = {
    "env": {
        "browser": true,
        "es2021": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 12,
        "sourceType": "module"
    },
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
    }
};
```

```TS
/* works as expected */
interface Transformer {
  apply(input: number): string;
}

const impl: Transformer = {
  apply(input) {
    return input.toString();
  },
};

/* produces unexpected warnings */
interface NestedTransformer {
  nested: {
    apply(input: number): string;
  };
}

const nestedImpl: NestedTransformer = {
  nested: {
    // Missing return type on function.eslint@typescript-eslint/explicit-module-boundary-types
    // Argument 'input' should be typed.eslint@typescript-eslint/explicit-module-boundary-types
    apply(input) {
      return input.toString();
    },
  },
};

export default {
  impl,
  nestedImpl,
};

```

**Expected Result**
There are no warnings for the nested apply function.

**Actual Result**
There are warnings for the nested apply function.

The following warnings were produced, even though the object is fully typed using an interface:
- Missing return type on function.eslint@typescript-eslint/explicit-module-boundary-types
- Argument 'input' should be typed.eslint@typescript-eslint/explicit-module-boundary-types

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.12.0` |
| `@typescript-eslint/parser`        | `4.12.0` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `7.17.0` |
| `node`                             | `12.19.1` |

**Comment by ccaspers** [comment]
Sorry I should have used the search function before posting this. It's aduplicate of https://github.com/typescript-eslint/typescript-eslint/issues/2864
